chingu-voyages/v37-bears-team-14
# Implement add/remove members API

## Change

Implements a project members API. The implementation uses a transaction session to rollback changes if the roleName change or member removal results in the project losing its one remaining owner.

```
GET    /api/v1/projects/:id/members
POST   /api/v1/projects/:id/members
DELETE /api/v1/projects/:id/members
```

## Testing

- Create a project
- Add a member
- Remove a member
- Check that these calls succeed

Example fetch calls:
```
fetch("/api/v1/projects/6213388bc787883b80dec9e7/members", {
    method:"DELETE",
    headers: {'content-type':'application/json'},
    body: JSON.stringify({ user: "6203839ca00bb00582d9586a" })
}).then(d=>d.json()).then(console.log)

fetch("/api/v1/projects/6213388bc787883b80dec9e7/members", {
    method:"POST",
    headers: {'content-type':'application/json'},
    body: JSON.stringify({ user: "62041c51a00bb005822d8388", roleName: "developer" })
}).then(d=>d.json()).then(console.log)
```

Closes #46
